src/Dfe.PlanTech.Application/Submission/Commands/SubmitAnswerCommand.cs

```diff
@@ -80,8 +83,12 @@ public async Task<bool> NextQuestionIsAnswered(int submissionId, string nextQues
 
                 if (submission != null && !submission.Completed)
                 {
-                    Domain.Questionnaire.Models.Question? nextUnansweredQuestion = await _GetNextUnansweredQuestion(submission.Id);
-                    return nextUnansweredQuestion != null ? (nextUnansweredQuestion, submission) : (null, submission);
+                    var nextUnansweredQuestion = await _GetNextUnansweredQuestion(submission.Id);
+                    var section = await _getSectionQuery.GetSectionById(sectionId, CancellationToken.None);
+
+                    if (nextUnansweredQuestion == null) return (null, submission);
```

Comment:
Should this be swapped with line 87? We don't appear to line 87 at this point, and this is a return statement, so we could be retrieving the section unnecessarily.

Reply:
Resolved in: [f297f43](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/288/commits/f297f43ca05efbf05f362162f529ed97376ec3a5)